services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_insights/history_and_recommendations/individual_student.tsx

```diff
@@ -0,0 +1,182 @@
+
+  const renderPracticeCard = (activities, relevantDiagnostics=null) => {
+    return (
+      <IndividualStudentPracticeCard
+        activities={activities}
+        relevantDiagnostic={relevantDiagnostics ? relevantDiagnostics[0] : null}
+      />
+    )
+  }
+
+  const renderStudentLearningSequenceRows = () => {
+    return DIAGNOSTIC_LEVELS.map((diagnosticLevel => {
+      const studentLearningSequencesForLevel = studentHistory.student_learning_sequences[diagnosticLevel]
+      if (!studentLearningSequencesForLevel?.length) { return null }
+
+      return studentLearningSequencesForLevel.map((studentLearningSequence) => {
+        const { id, classroom_id, student_learning_sequence_activities, } = studentLearningSequence
+        const { pre_diagnostic, post_diagnostic, pre_recommendation, post_recommendation, } = student_learning_sequence_activities
+        return (
+          <section className="student-learning-sequence" key={id}>
+            <div className="top-row">
+              <h4>{titleCase(diagnosticLevel)} Diagnostic</h4>
+              {teacherClassroomIds.includes(classroom_id) ? 'Assigned by you' : 'Assigned by another teacher'}
+            </div>
+            <div className="student-learning-sequence-cards">
+              {renderDiagnosticCard(pre_diagnostic)}
+              {renderPracticeCard(pre_recommendation, pre_diagnostic)}
+              {renderDiagnosticCard(post_diagnostic, pre_diagnostic)}
+              {renderPracticeCard(post_recommendation, post_diagnostic)}
+            </div>
+          </section>
+        )
+      })
+    }))
+  }
+
+  return (
+    <section className="diagnostic-insights-content history-and-recommendations by-student">
+      <section className="individual-student-recommendations">
+        <h3>
+          <span>
+            <img alt="" src={`${SMALL_ICON_BASE_SRC}/star.svg`} />
+            Diagnostic Recommendation
+          </span>
+          <span className="timeframe">{timeframe}</span>
+        </h3>
+        <section className="individual-student-recommendations-container">
+          <div className="top-line">
+            <p>{recommendationExplainer()}</p>
+            <div>
+              <span>Was this recommendation helpful?</span>
+              <button aria-label="Submit feedback about this recommendation" className="quill-button outlined small grey icon focus-on-light" type="button">
+                <img alt="" src={`${EXTRA_SMALL_ICON_BASE_SRC}/thumbs-up.svg`} />
+                <img alt="" src={`${EXTRA_SMALL_ICON_BASE_SRC}/thumbs-down.svg`} />
+              </button>
+            </div>
+          </div>
+          <div className="recommendations-wrapper">
+            {DIAGNOSTIC_LEVELS.map(diagnosticLevel => renderRecommendationCard(diagnosticLevel))}
+          </div>
+        </section>
+      </section>
+      <section className="individual-student-history">
+        <h3>
+          <span>
+            <img alt="" src={`${SMALL_ICON_BASE_SRC}/clock.svg`} />
```

Comment:
let's maybe hoist all of these src strings in this return statement to the top as consts to stay consistent with other files?